bumpversion/cli.py

```diff
@@ -89,233 +126,225 @@ def split_args_in_optional_and_positional(args):
     )
-
-    parser1.add_argument(
+    root_parser.add_argument(
         "--allow-dirty",
         action="store_true",
         default=False,
         help="Don't abort if working directory is dirty",
         required=False,
     )
+    known_args, _ = root_parser.parse_known_args(args)
+    return args, known_args, root_parser, positionals
 
-    known_args, remaining_argv = parser1.parse_known_args(args)
 
+def _setup_logging(show_list, verbose):
     logformatter = logging.Formatter("%(message)s")
-
     if not logger_list.handlers:
         ch2 = logging.StreamHandler(sys.stdout)
         ch2.setFormatter(logformatter)
         logger_list.addHandler(ch2)
-
-    if known_args.list:
+    if show_list:
         logger_list.setLevel(logging.DEBUG)
-
     try:
-        log_level = [logging.WARNING, logging.INFO, logging.DEBUG][known_args.verbose]
+        log_level = [logging.WARNING, logging.INFO, logging.DEBUG][verbose]
     except IndexError:
         log_level = logging.DEBUG
-
     root_logger = logging.getLogger('')
     root_logger.setLevel(log_level)
-
     logger.debug("Starting %s", DESCRIPTION)
 
-    defaults = {}
-    vcs_info = {}
 
+def _determine_vcs_usability():
+    vcs_info = {}
     for vcs in VCS:
         if vcs.is_usable():
             vcs_info.update(vcs.latest_tag_info())
+    return vcs_info
+
 
+def _determine_current_version(vcs_info):
+    defaults = {}
     if "current_version" in vcs_info:
         defaults["current_version"] = vcs_info["current_version"]
+    return defaults
 
-    explicit_config = hasattr(known_args, "config_file")
 
+def _determine_config_file(explicit_config):
     if explicit_config:
-        config_file = known_args.config_file
-    elif not os.path.exists(".bumpversion.cfg") and os.path.exists("setup.cfg"):
-        config_file = "setup.cfg"
-    else:
-        config_file = ".bumpversion.cfg"
+        return explicit_config
+    if not os.path.exists(".bumpversion.cfg") and os.path.exists("setup.cfg"):
+        return "setup.cfg"
+    return ".bumpversion.cfg"
 
+
+def _load_configuration(config_file, explicit_config, defaults):
     # setup.cfg supports interpolation - for compatibility we must do the same.
     if os.path.basename(config_file) == "setup.cfg":
         config = ConfigParser("")
     else:
         config = RawConfigParser("")
-
     # don't transform keys to lowercase (which would be the default)
     config.optionxform = lambda option: option
-
     config.add_section("bumpversion")
-
     config_file_exists = os.path.exists(config_file)
 
-    part_configs = {}
+    if not config_file_exists:
+        message = "Could not read config file at {}".format(config_file)
+        if explicit_config:
+            raise argparse.ArgumentTypeError(message)
+        logger.info(message)
+        return config, config_file_exists, None, {}, []
 
-    files = []
+    logger.info("Reading config file %s:", config_file)
+    # TODO: this is a DEBUG level log
```

Comment:
Doesn't this belong to the logger.info below on line 237?

Reply:
True